lib/indico.rb

```diff
@@ -183,50 +81,90 @@ def self.image_recognition(image, config = nil)
     api_handler(preprocess(image, 144, true), 'imagerecognition', config)
   end
 
-  def self.batch_image_features(image, config = nil)
-    warn(
-      "The `batch_image_features` function will be deprecated in the next major upgrade. " +
-      "Please call `image_features` instead with the same arguments"
-    )
-    self.image_features(image, config)
-  end
-
   def self.content_filtering(image, config = nil)
     api_handler(preprocess(image, 128, true), 'contentfiltering', config)
   end
 
-  def self.batch_content_filtering(image, config = nil)
-    warn(
-      "The `batch_content_filtering` function will be deprecated in the next major upgrade. " +
-      "Please call `content_filtering` instead with the same arguments"
-    )
-    self.content_filtering(image, config)
-  end
-
   def self.analyze_image(face, apis = IMAGE_APIS, config = nil)
     api_hash = {apis:apis}
     multi(preprocess(face, 48, false), "image", apis, IMAGE_APIS, config ? config.update(api_hash) : api_hash)
   end
 
-  def self.batch_analyze_image(image, apis = IMAGE_APIS, config = nil)
-    warn(
-      "The `batch_analyze_image` function will be deprecated in the next major upgrade. " +
-      "Please call `analyze_image` instead with the same arguments"
-    )
-    self.analyze_image(image, apis, config)
-  end
-
   def self.analyze_text(text, apis = TEXT_APIS, config = nil)
     api_hash = {apis:apis}
     multi(text, "text", apis, TEXT_APIS, config ? config.update(api_hash) : api_hash)
   end
 
-  def self.batch_analyze_text(text, apis = TEXT_APIS, config = nil)
-    warn(
-      "The `batch_analyze_text` function will be deprecated in the next major upgrade. " +
-      "Please call `analyze_text` instead with the same arguments"
-    )
-    self.analyze_text(text, apis, config)
+  def self.collections(config = nil)
+    Indico.api_handler(nil, 'custom', config, 'collections')
+  end
+
+  class Collection
+
+      def initialize(collection)
+        @collection = collection
+      end
+
+      def add_data(data, config = nil)
+
+        batch = data[0].kind_of?(Array)
+        if batch
+          x, y = data.transpose
+          x = Indico::preprocess(x, 144, true)
+          data = x.zip(y)
+        else
+          data[0] = Indico::preprocess(data[0], 144, true)
+        end
+
+        if config.nil?
+          config = Hash.new()
+        end
+        config[:collection] = @collection
+        Indico.api_handler(data, 'custom', config, 'add_data')
+      end
+
+      def train(config = nil)
+        if config.nil?
+          config = Hash.new()
+        end
+        config[:collection] = @collection
+        Indico.api_handler(nil, 'custom', config, 'train')
+      end
+
+      def wait(interval = 1)
+        while info()['status'] != "ready" do
+          sleep(interval)
+        end
+      end
+
+      def info(interval = 1)
+        Indico.collections()[@collection]
```

Comment:
Although I feel like the standard typeerror raised should be sufficient to let people know what went wrong?